utils/s3/client.cc

```diff
@@ -92,7 +94,9 @@ client::client(std::string host, endpoint_config_ptr cfg, semaphore& mem, global
         , _retry_strategy(std::move(rs)) {
     _creds_provider_chain
         .add_credentials_provider(std::make_unique<aws::config_file_aws_credentials_provider>(db::config::get_conf_sub("object_storage.yaml").native()))
-        .add_credentials_provider(std::make_unique<aws::environment_aws_credentials_provider>());
+        .add_credentials_provider(std::make_unique<aws::environment_aws_credentials_provider>())
+        .add_credentials_provider(std::make_unique<aws::sts_assume_role_credentials_provider>(_cfg->region, _cfg->role_arn.value_or("")))
```

Comment:
What's the point in making config::role_arn optional if the code converts it to empty string if not set anyway?